tests/nfs_server_ns.go

```diff
@@ -0,0 +1,259 @@
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	deploy "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/apps/v1/deployment"
+	container "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/core/v1/container"
+	pvc "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/core/v1/persistentvolumeclaim"
+	pts "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/core/v1/podtemplatespec"
+	volume "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/core/v1/volume"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var _ = Describe("TEST CUSTOM NAMESPACE FOR NFS SERVER", func() {
+	var (
+		accessModes          = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteMany}
+		capacity             = "2Gi"
+		deployName           = "busybox-nfs"
+		label                = "demo=nfs-deployment"
+		pvcName              = "pvc-nfs"
+		nfsServerNsEnv       = "OPENEBS_IO_NFS_SERVER_NS"
+		OpenEBSNamespace     = "openebs"
+		NFSProvisionerName   = "openebs-nfs-provisioner"
+		applicationNamespace = "nfs-tests-ns"
+		labelselector        = map[string]string{
+			"demo": "nfs-deployment",
+		}
+	)
+
+	When("provisioner deployment updated with NFSServerNamespace", func() {
+		It("should update the provisioner deployment", func() {
+			By("updating a deployment")
+			deploy, err := Client.getDeployment(OpenEBSNamespace, NFSProvisionerName)
+			Expect(err).To(
+				BeNil(),
+				"while fetching deployment {%s} in namespace {%s}",
+				NFSProvisionerName,
+				OpenEBSNamespace,
+			)
+
+			By("updating the deployment")
+			nsEnv := corev1.EnvVar{
+				Name:  nfsServerNsEnv,
+				Value: applicationNamespace,
+			}
+
+			deploy.Spec.Template.Spec.Containers[0].Env = append(
+				deploy.Spec.Template.Spec.Containers[0].Env,
+				nsEnv,
+			)
+			_, err = Client.updateDeployment(deploy)
+			Expect(err).To(
+				BeNil(),
+				"while updating deployment {%s} in namespace {%s}",
+				NFSProvisionerName,
+				OpenEBSNamespace,
+			)
+			By("waiting for deployment rollout")
+			err = Client.waitForDeploymentRollout(OpenEBSNamespace, NFSProvisionerName)
+			Expect(err).To(BeNil(), "while verifying deployment rollout")
+		})
+	})
+
+	When("pvc with storageclass openebs-rwx is created", func() {
+		It("should create a pvc ", func() {
+			var (
+				scName = "openebs-rwx"
+			)
+
+			By("building a pvc")
+			pvcObj, err := pvc.NewBuilder().
+				WithName(pvcName).
+				WithNamespace(applicationNamespace).
+				WithStorageClass(scName).
+				WithAccessModes(accessModes).
+				WithCapacity(capacity).Build()
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while building pvc {%s} in namespace {%s}",
+				pvcName,
+				applicationNamespace,
+			)
+
+			By("creating above pvc")
+			err = Client.createPVC(pvcObj)
+			Expect(err).To(
+				BeNil(),
+				"while creating pvc {%s} in namespace {%s}",
+				pvcName,
+				applicationNamespace,
+			)
```

Comment:
Can we verify PVC bound status before moving to next case? If there is some delay in updating PV name on PVC then next when will fail

Reply:
`createPVC` ensure bounding